sdk/cmake/infiles/fpga-app.sh.in

```diff
@@ -0,0 +1,58 @@
+# !/bin/bash
+
+set -e
+
+if [[ -z "${FP_PATH}" ]]; then
+    echo "Environment variable FP_PATH not set!"
+    exit 1
+fi
+
+if [[ -z "${FP_SDK_PATH}" ]]; then
+    echo "Environment variable FP_SDK_PATH not set!"
+    exit 1
+fi
+
+flashdev=@FP_FLASH_DEVICE@
+baudrate=@UART_BAUDRATE@
+imem_path=@IMEM_LOCATION@/@IMEM_NAME@
+imem_serialized_path=/tmp/@IMEM_NAME@.serialized
+interface_cmd_sourcefile="${FP_SDK_PATH}/cmake/utils/fpga-interface/@FP_INTERFACE_PROGRAM@.source"
+
+echo "Serializing app and placing its result in /tmp"
+python3 ${FP_PATH}/scripts/serialize_app.py ${imem_path} ${imem_serialized_path}
+
+echo "Flashing serialized app on device ${flashdev} with baudrate ${baudrate}"
+
+# Script takes usb_path, baudrate, and file
+python3 ${FP_PATH}/scripts/send_uart.py ${flashdev} ${baudrate} ${imem_serialized_path}
+
+if [[ -z "${FP_SDK_FPGA_INTERFACE_PROGRAM}" ]]; then
+    exit 0
+fi
+# Source the user selected program to interface with the FPGA, e.g., `picocom.source`
+# This sets the `interface_cmd` variable
+source "$interface_cmd_sourcefile"
+
+# Print out info on how to interface with software
+echo "To interface with the FPGA, run:"
+echo "$interface_cmd"
+
+# Check if `xclip` exists
+if ! command -v xclip > /dev/null 2 >& 1
```

Comment:
nice touch

Reply:
Haha thank you! :) It was quite fun to tinker around with this